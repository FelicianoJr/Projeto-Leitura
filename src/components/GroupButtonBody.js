import React from "react";
import PropTypes from "prop-types";

const GroupButtonBody = ({ edit, remove, voteUp, voteDown }) => {
  return (
    <div>
      <button
        id="edit"
        type="button"
        className="btn btn-light btn-sm mr-sm-2"
        onClick={edit}
      >
        Editar
      </button>
      <button
        id="delete"
        type="button"
        className="btn btn-light btn-sm mr-sm-2"
        onClick={remove}
      >
        Excluir
      </button>
      <button
        id="like"
        type="button"
        className="btn btn-light btn-sm mr-sm-2"
        onClick={voteUp}
      >
        Gostei
      </button>
      <button
        id="dislike"
        type="button"
        className="btn btn-light btn-sm"
        onClick={voteDown}
      >
        NÃ£o Gostei
      </button>
    </div>
  );
};

GroupButtonBody.protoTypes = {
  showComment: PropTypes.func,
  newComment: PropTypes.func,
  edit: PropTypes.func,
  remove: PropTypes.func,
  voteUp: PropTypes.func,
  voteDown: PropTypes.func
};

export default GroupButtonBody;
