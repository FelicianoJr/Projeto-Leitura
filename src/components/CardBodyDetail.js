import React from "react";
import PropTypes from "prop-types";
import { format } from "date-fns";
import GroupButtonBody from "./GroupButtonBody";

const CardBodyDetail = ({ data, edit, remove, voteDown, voteUp }) => {
  return (
    <div className="card-body  text-dark">
      {data.title && <div className="card-title">{data.title}</div>}
      <div className="card-text">{data.body}</div>
      <div className="card-text">
        <small>
          <b>Pontuação:</b>
        </small>{" "}
        <div className="badge">{data.voteScore}</div>
      </div>
      <small className="text-muted ">
        Criado por {data.author} em {format(data.timestamp, "DD/MM/YYYY")}
      </small>

      <GroupButtonBody
        count={data.commentCount}
        edit={edit}
        remove={remove}
        voteDown={voteDown}
        voteUp={voteUp}
      />
    </div>
  );
};

CardBodyDetail.propTypes = {
  title: PropTypes.string,
  body: PropTypes.string,
  author: PropTypes.string,
  timestamp: PropTypes.number,
  voteScore: PropTypes.number
};

export default CardBodyDetail;
