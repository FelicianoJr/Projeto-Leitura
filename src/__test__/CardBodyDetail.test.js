import React from "react";
import { shallow } from "enzyme";
import CardBodyDetail from "../components/CardBodyDetail";
import GroupButtonBody from "../components/GroupButtonBody";

const setup = data => {
  const component = shallow(<CardBodyDetail data={data} />);

  return {
    component: component,
    divTitle: component.find(".card-title"),
    divBody: component.find(".card-text"),
    badgeScore: component.find(".badge"),
    b: component.find("b"),
    small: component.find("small"),
    groupButton: component.find(GroupButtonBody)
  };
};

describe("<CardBodyDetail/>", () => {
  let postProps;
  beforeEach(() => {
    postProps = {
      title: "Udacity",
      body: "new situation",
      voteScore: 4,
      author: "Feliciano",
      timestamp: 1467166872634
    };
  });

  it("should render all Detail", () => {
    const { divTitle, divBody, badgeScore, small } = setup(postProps);
    expect(divTitle.text()).toEqual("Udacity");
    expect(divBody.at(0).text()).toEqual("new situation");
    expect(badgeScore.text()).toEqual("4");
    expect(small.at(1).text()).toEqual("Criado por Feliciano em 28/06/2016");
  });

  it("render one <GroupButtonCard/>", () => {
    const { groupButton } = setup(postProps);
    expect(groupButton.length).toEqual(1);
  });

  it("snapshot  a cardDetail", () => {
    const { component } = setup(postProps);
    expect(component).toMatchSnapshot();
  });
});
