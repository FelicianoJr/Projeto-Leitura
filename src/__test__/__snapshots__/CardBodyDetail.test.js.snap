// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardBodyDetail/> snapchot  a cardDetail 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBodyDetail
    data={
        Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 1467166872634,
            "title": "Udacity",
            "voteScore": 4,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          Udacity
</div>,
        <div
          className="card-text"
>
          new situation
</div>,
        <div
          className="card-text"
>
          <small>
                    <b>
                              Pontuação:
                    </b>
          </small>
           
          <div
                    className="badge"
          >
                    4
          </div>
</div>,
        <small
          className="text-muted "
>
          Criado por 
          Feliciano
           em 
          28/06/2016
</small>,
        <GroupButtonBody
          count={undefined}
          edit={undefined}
          remove={undefined}
          voteDown={undefined}
          voteUp={undefined}
/>,
      ],
      "className": "card-body  text-dark",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Udacity",
          "className": "card-title",
        },
        "ref": null,
        "rendered": "Udacity",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "new situation",
          "className": "card-text",
        },
        "ref": null,
        "rendered": "new situation",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <small>
              <b>
                            Pontuação:
              </b>
</small>,
            " ",
            <div
              className="badge"
>
              4
</div>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                Pontuação:
</b>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pontuação:",
              },
              "ref": null,
              "rendered": "Pontuação:",
              "type": "b",
            },
            "type": "small",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "badge",
            },
            "ref": null,
            "rendered": 4,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Criado por ",
            "Feliciano",
            " em ",
            "28/06/2016",
          ],
          "className": "text-muted ",
        },
        "ref": null,
        "rendered": Array [
          "Criado por ",
          "Feliciano",
          " em ",
          "28/06/2016",
        ],
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "count": undefined,
          "edit": undefined,
          "remove": undefined,
          "voteDown": undefined,
          "voteUp": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            Udacity
</div>,
          <div
            className="card-text"
>
            new situation
</div>,
          <div
            className="card-text"
>
            <small>
                        <b>
                                    Pontuação:
                        </b>
            </small>
             
            <div
                        className="badge"
            >
                        4
            </div>
</div>,
          <small
            className="text-muted "
>
            Criado por 
            Feliciano
             em 
            28/06/2016
</small>,
          <GroupButtonBody
            count={undefined}
            edit={undefined}
            remove={undefined}
            voteDown={undefined}
            voteUp={undefined}
/>,
        ],
        "className": "card-body  text-dark",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Udacity",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "Udacity",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "new situation",
            "className": "card-text",
          },
          "ref": null,
          "rendered": "new situation",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <small>
                <b>
                                Pontuação:
                </b>
</small>,
              " ",
              <div
                className="badge"
>
                4
</div>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  Pontuação:
</b>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pontuação:",
                },
                "ref": null,
                "rendered": "Pontuação:",
                "type": "b",
              },
              "type": "small",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "badge",
              },
              "ref": null,
              "rendered": 4,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Criado por ",
              "Feliciano",
              " em ",
              "28/06/2016",
            ],
            "className": "text-muted ",
          },
          "ref": null,
          "rendered": Array [
            "Criado por ",
            "Feliciano",
            " em ",
            "28/06/2016",
          ],
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": undefined,
            "edit": undefined,
            "remove": undefined,
            "voteDown": undefined,
            "voteUp": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<CardBodyDetail/> snapshot  a cardDetail 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardBodyDetail
    data={
        Object {
            "author": "Feliciano",
            "body": "new situation",
            "timestamp": 1467166872634,
            "title": "Udacity",
            "voteScore": 4,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card-title"
>
          Udacity
</div>,
        <div
          className="card-text"
>
          new situation
</div>,
        <div
          className="card-text"
>
          <small>
                    <b>
                              Pontuação:
                    </b>
          </small>
           
          <div
                    className="badge"
          >
                    4
          </div>
</div>,
        <small
          className="text-muted "
>
          Criado por 
          Feliciano
           em 
          28/06/2016
</small>,
        <GroupButtonBody
          count={undefined}
          edit={undefined}
          remove={undefined}
          voteDown={undefined}
          voteUp={undefined}
/>,
      ],
      "className": "card-body  text-dark",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Udacity",
          "className": "card-title",
        },
        "ref": null,
        "rendered": "Udacity",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "new situation",
          "className": "card-text",
        },
        "ref": null,
        "rendered": "new situation",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <small>
              <b>
                            Pontuação:
              </b>
</small>,
            " ",
            <div
              className="badge"
>
              4
</div>,
          ],
          "className": "card-text",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <b>
                Pontuação:
</b>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Pontuação:",
              },
              "ref": null,
              "rendered": "Pontuação:",
              "type": "b",
            },
            "type": "small",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 4,
              "className": "badge",
            },
            "ref": null,
            "rendered": 4,
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Criado por ",
            "Feliciano",
            " em ",
            "28/06/2016",
          ],
          "className": "text-muted ",
        },
        "ref": null,
        "rendered": Array [
          "Criado por ",
          "Feliciano",
          " em ",
          "28/06/2016",
        ],
        "type": "small",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "count": undefined,
          "edit": undefined,
          "remove": undefined,
          "voteDown": undefined,
          "voteUp": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card-title"
>
            Udacity
</div>,
          <div
            className="card-text"
>
            new situation
</div>,
          <div
            className="card-text"
>
            <small>
                        <b>
                                    Pontuação:
                        </b>
            </small>
             
            <div
                        className="badge"
            >
                        4
            </div>
</div>,
          <small
            className="text-muted "
>
            Criado por 
            Feliciano
             em 
            28/06/2016
</small>,
          <GroupButtonBody
            count={undefined}
            edit={undefined}
            remove={undefined}
            voteDown={undefined}
            voteUp={undefined}
/>,
        ],
        "className": "card-body  text-dark",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Udacity",
            "className": "card-title",
          },
          "ref": null,
          "rendered": "Udacity",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "new situation",
            "className": "card-text",
          },
          "ref": null,
          "rendered": "new situation",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <small>
                <b>
                                Pontuação:
                </b>
</small>,
              " ",
              <div
                className="badge"
>
                4
</div>,
            ],
            "className": "card-text",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  Pontuação:
</b>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Pontuação:",
                },
                "ref": null,
                "rendered": "Pontuação:",
                "type": "b",
              },
              "type": "small",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 4,
                "className": "badge",
              },
              "ref": null,
              "rendered": 4,
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Criado por ",
              "Feliciano",
              " em ",
              "28/06/2016",
            ],
            "className": "text-muted ",
          },
          "ref": null,
          "rendered": Array [
            "Criado por ",
            "Feliciano",
            " em ",
            "28/06/2016",
          ],
          "type": "small",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "count": undefined,
            "edit": undefined,
            "remove": undefined,
            "voteDown": undefined,
            "voteUp": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
